mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp

```diff
@@ -21,7 +21,12 @@ using namespace mlir;
 //===----------------------------------------------------------------------===//
 
 LogicalResult GENX::MatrixDPASOp::verify() {
-  // TODO
+  if (getSd() != 8)
+    return this->emitOpError("expecting systolic depth to be 8");
```

Comment:
This makes me think... Given that SD must be 8 there isn't any good reason for the operation to have the attribute in the first place.  Looking at the DPAS instruction description (https://github.com/intel/intel-graphics-compiler/blob/master/documentation/visa/instructions/DPAS.md) I found:
```
SD can be encoded as 1, 2, 4, and 8. XEHP+ only supports a systolic depth of 8.
```
I think the verification code should check that SD is 1, 2, 4, or 8. Is up to the producer of this operation to provide a valid SD for the platform they target IMO.

Reply:
Also you might want to update the description of the operation and add a blurb to indicate the constraint on SD and RC.

Reply:
Thinking about ExecSize, the instruction manual documents that ExecSize can be 8 or 16:
```
Exec_size(ub): Execution size

Bit[2..0]: size of the region for source and destination operands

{XEHP}0b011: 8 elements
{PVC}0b100: 16 elements
```
I think to make the operation as generic as possible we should add that attr as well and verify it is either 8 or 16.
The producer will need to set it up correctly for the target architecture (so it will pass 16 on PVC).

Reply:
Another comment. The verification code should ensure that PA and PB have valid combination w.r.t the operand types. The supported combos are:
```
.. table:: All legal combinations of types and precisions.
  :align: center

  +--------+---------+-----------+----------+
  | Dst    |  Src0   | Src1      | Src2     |
  |        |         | Precision | Preision |
  +--------+---------+-----------+----------+
  | UD, D  |  UD,D   | int       | int      |
  +--------+---------+-----------+----------+
  | F, BF  |  F, BF  | BF        | BF       |
  +--------+---------+-----------+----------+
  | F, HF  |  F, HF  | HF        | HF       |
  +--------+---------+-----------+----------+
  | F      |  F      | TF32      | TF32     |
  +--------+---------+-----------+----------+
  | F      |  F      | BF8, HF8  | BF8, HF8 |
  +--------+---------+-----------+----------+

Reply:
Decided offline to restrict the operation to only support PVC for now.

Reply:
Added TODO, will address in next PR.